profiling/service/service.go

```diff
@@ -0,0 +1,78 @@
+/*
+ *
+ * Copyright 2019 gRPC authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package service
+
+import (
+	"context"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/grpclog"
+
+	"google.golang.org/grpc/internal/profiling"
+	ppb "google.golang.org/grpc/profiling/proto"
+	pspb "google.golang.org/grpc/profiling/proto/service"
+)
+
+type ProfilingConfig struct {
+	Enabled         bool
+	StreamStatsSize uint32
+	Server          *grpc.Server
+}
+
+func registerService(s *grpc.Server) {
+	grpclog.Infof("registering profiling service")
+	pspb.RegisterProfilingServer(s, &profilingServer{})
+}
+
+func Init(pc *ProfilingConfig) (err error) {
+	if err = profiling.InitStats(pc.StreamStatsSize); err != nil {
+		return
+	}
+
+	registerService(pc.Server)
```

Comment:
Do you need to define a function for this? It is just two lines, and I don't see it being called from other places.

Reply:
Correct, done.